model/spectrum.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * spectrum.h
+ */
+#ifndef DXTBX_MODEL_SPECTRUM_H
+#define DXTBX_MODEL_SPECTRUM_H
+
+#include <iostream>
+#include <cmath>
+#include <scitbx/vec3.h>
+#include <scitbx/array_family/shared.h>
+#include <scitbx/array_family/simple_io.h>
+#include <scitbx/array_family/simple_tiny_io.h>
+#include <dxtbx/error.h>
+#include "model_helpers.h"
+
+namespace dxtbx { namespace model {
+  typedef scitbx::af::shared<double> vecd;
+
+  /** A class to represent a 2D spectrum. */
+  class Spectrum {
+  public:
+    /** Default constructor: initialise all to zero */
+    Spectrum() {}
+
+    /**
+     * Initialise all the spectrum parameters.
+     * @param energies The spectrum energies (eV)
+     * @param energies The spectrum weights (unitless)
+     */
+    Spectrum(vecd energies, vecd weights)
+        : energies_(energies),
+          weights_(weights) {}
+
+    virtual ~Spectrum() {}
+
+    /* Get the spectrum energies (eV) */
+    vecd get_energies() const {
+      return energies_;
+    }
+
+    /* Get the spectrum weights (unitless) */
+    vecd get_weights() const {
+      return weights_;
+    }
+
+    double get_weighted_wavelength() const {
+      if (energies_.size() == 0)
+        return 0;
+      double weighted_sum = 0;
```

Comment:
Since this is a static data table, worth computing these in the constructor and saving the values? That way if (just sayin') someone were to pass in nominal values into some theoretical constructor then they could be recovered without having to create a data array to recompute the values from. Would also mean calling ` `get_weighted_wavelength` could be treated compute wise as `get_wavelength()` without excessive compute overhead.

Reply:
Right, in our use case we aren't even using get_weighted_wavelength at all, it's just a convenience function.  We are using the channels and energies directly.  I would rather not compute the weighted sums if they won't be used.

Reply:
Gonna keep this discussion open, I think there is a lot of merit in computing some metrics in the constructor which would be real useful down the line when it comes to treating the data in other ways (e.g. pink beam)